astropy/nddata/utils.py

```diff
@@ -916,3 +925,373 @@ def bbox_cutout(self):
         are for the valid (non-filled) cutout values.
         """
         return self._calc_bbox(self.slices_cutout)
+
+
+def make_cutouts(data, catalog, wcs=None, origin=0, verbose=True):
+    """Make cutouts of catalog targets from a 2D image array.
+
+    Notes
+    -----
+    The input Catalog must have the following columns, which must have
+    `~astropy.unit.Unit`s where applicable:
+
+        * ``'id'`` - ID string; no unit necessary.
+        * ``'coords'`` - SkyCoord (Overrides ra, dec, x and y columns).
+        * ``'ra'`` or ``'x'``- RA (angular units e.g., deg, H:M:S, arcsec etc..)
+          or pixel x position (only in `~astropy.units.pix`).
+        * ``'dec'`` or ``'y'`` - Dec (angular units e.g., deg, D:M:S, arcsec etc..)
+          or pixel y position (only in `~astropy.units.pix`).
+        * ``'cutout_width'`` - Cutout width (e.g., in arcsec, pix).
+        * ``'cutout_height'`` - Cutout height (e.g., in arcsec, pix).
+
+    Optional columns:
+        * ``'cutout_pa'`` - Cutout angle (e.g., in deg, arcsec). This is only
+          use if user chooses to rotate the cutouts. Positive value
+          will result in a clockwise rotation.
+
+    If saved to fits, cutouts are organized as follows:
+        <output_dir>/
+            <id>.fits
+
+    Each cutout image is a simple single-extension FITS with updated WCS.
+    Its header has the following special keywords:
+
+        * ``OBJ_RA`` - RA of the cutout object in degrees.
+        * ``OBJ_DEC`` - DEC of the cutout object in degrees.
+        * ``OBJ_ROT`` - Rotation of cutout object in degrees.
+
+    Parameters
+    ----------
+    data : 2D `~numpy.ndarray` or `~astropy.nddata.NDData`
+        The 2D cutout array.
+    catalog : `~astropy.table.table.Table`
+        Catalog table defining the sources to cut out. Must contain
+        unit information as the cutouttool does not assume default units.
+    wcs : `~astropy.wcs.wcs.WCS`
+        WCS if the input image is `~numpy.ndarray`.
+    origin : int
+        Whether SkyCoord.from_pixel should use 0 or 1-basedpixel coordinates.
+    verbose : bool
+        Print extra info. Default is `True`.
+
+    Returns
+    -------
+    cutouts : list
+        A list of NDData. If cutout failed for a target,
+       `None` will be added as a place holder.
+    """
+    # Optional dependencies...
+    try:
+        from reproject.interpolation.high_level import reproject_interp
+    except ImportError as e:
+        raise ImportError("Optional requirement not met: " + e.msg)
```

Comment:
I would probably check for this dependency only if `reproject` is actually needed (i.e. if rotation was requested; `apply_rotation=True`)